src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -1623,17 +1640,20 @@ class WindowMergeTask : public ExecutorTask {
 
 TaskExecutionResult WindowMergeTask::ExecuteTask(TaskExecutionMode mode) {
 	// Loop until all hash groups are done
-	while (hash_groups.completed != hash_groups.states.size()) {
+	size_t sorted = 0;
+	while (sorted < hash_groups.states.size()) {
 		// First check if there is an unfinished task for this thread
 		if (!local_state.TaskFinished()) {
 			local_state.ExecuteTask();
 			continue;
 		}
 
 		// Thread is done with its assigned task, try to fetch new work
-		for (auto &global_state : hash_groups.states) {
+		for (auto group = sorted; group < hash_groups.states.size(); ++group) {
```

Comment:
Does it always work to start the loop at `group = sorted`?

What if the first hash group is large, let's say it can occupy 3 threads, and the second hash group is small; it can occupy 1 thread. Assuming we have 4 threads, a thread will start working on the second hash group before the first hash group is completed.

The second hash group is small enough that it will be completed before the first hash group. Now, `sorted` is updated from `0` to `1`, and we will never complete sorting the first hash group.

Does this make sense? Or perhaps I am overlooking something.